File: JAVA-GEOMETRIA/Elipse.java
import java.awt.*;

public class Elipse {
    private final double centroX;
    private final double centroY;
    private final double ancho;
    private final double alto;
    private final Color borde;
    private final Color relleno;

    public Elipse(double centroX, double centroY, double ancho, double alto, Color borde, Color relleno) {
        this.centroX = centroX;
        this.centroY = centroY;
        this.ancho = ancho;
        this.alto = alto;
        this.borde = borde;
        this.relleno = relleno;
    }

    public double calcularArea() {
        return Math.PI * (ancho / 2) * (alto / 2);
    }

    public double calcularPerimetro() {
        return Math.PI * (3 * (ancho / 2 + alto / 2) - Math.sqrt((3 * (ancho / 2) + (alto / 2)) * ((ancho / 2) + 3 * (alto / 2))));
    }

    public void dibujar(Graphics g) {
        g.setColor(relleno);
        g.fillOval((int) (centroX - ancho / 2), (int) (centroY - alto / 2), (int) ancho, (int) alto);
        g.setColor(borde);
        g.drawOval((int) (centroX - ancho / 2), (int) (centroY - alto / 2), (int) ancho, (int) alto);
        g.drawString("Área: " + String.format("%.2f", calcularArea()), (int) centroX, (int) centroY + (int) (alto / 2) + 15);
        g.drawString("Perímetro: " + String.format("%.2f", calcularPerimetro()), (int) centroX, (int) centroY + (int) (alto / 2) + 30);
    }
}

File: CSHARDP-GEOMETRIA/Elipse.cs
using System.Drawing;

namespace CSHARDP_GEOMETRIA
{
    public class Elipse
    {
        private float Cx; // Centro X
        private float Cy; // Centro Y
        private float Ancho;
        private float Alto;
        private Color Borde;
        private Color Relleno;

        public Elipse(float cx, float cy, float ancho, float alto, Color borde, Color relleno)
        {
            Cx = cx;
            Cy = cy;
            Ancho = ancho;
            Alto = alto;
            Borde = borde;
            Relleno = relleno;
        }

        public float CalcularArea()
        {
            return (float)(Math.PI * (Ancho / 2) * (Alto / 2));
        }

        public float CalcularPerimetro()
        {
            return (float)(Math.PI * (3 * (Ancho / 2 + Alto / 2) - Math.Sqrt((3 * (Ancho / 2) + (Alto / 2)) * ((Ancho / 2) + 3 * (Alto / 2)))));
        }

        public void Dibujar(Graphics g)
        {
            using (Brush brush = new SolidBrush(Relleno))
            {
                g.FillEllipse(brush, Cx - Ancho / 2, Cy - Alto / 2, Ancho, Alto);
            }
            using (Pen pen = new Pen(Borde))
            {
                g.DrawEllipse(pen, Cx - Ancho / 2, Cy - Alto / 2, Ancho, Alto);
            }

            g.DrawString($"Área: {CalcularArea():F2}", SystemFonts.DefaultFont, Brushes.Black, Cx, Cy + (Alto / 2) + 5);
            g.DrawString($"Perímetro: {CalcularPerimetro():F2}", SystemFonts.DefaultFont, Brushes.Black, Cx, Cy + (Alto / 2) + 20);
        }
    }
}
